typescript/looker/rtl/proxySession.ts

```diff
@@ -0,0 +1,68 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2019 Looker Data Sciences, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+import { AuthSession } from './authSession'
+import { BrowserTransport } from './browserTransport'
+import { ITransport, IRequestProps } from './transport'
+import { IApiSettings } from './apiSettings'
+
+/**
+ * An AuthSession class intended for use with proxied requests
+ *
+ * This session uses `authenticate` to modify all requests to pass through a proxy server with
+ * passing the request to the proxy server specified in the `proxyUrl` parameter, and sets
+ * the original request path as an `X-Forwarded-For` header.
+ *
+ */
+export abstract class ProxySession extends AuthSession {
```

Comment:
what else is needed for a subclass implementation? seems like `login()`, `logout()`, `getToken()` would do something like send the configured creds along to the proxy server? Or is the proxy server configured with the creds? If the latter, that seems really dangerous unless there's some network level access restrictions.

Reply:
There shouldn't be much, but I haven't actually implemented the proxy server to correspond to this proxy session class. The security would need to be implemented in the proxy server. The proxy session should never see or pass the credentials. I'm happy to iterate on this implementation as we learn that it needs to change. This is to get the proxy service started, at least.

Reply:
👍